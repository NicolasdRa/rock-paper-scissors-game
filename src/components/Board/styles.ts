import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    main: {
      marginTop: '2rem',
      position: 'relative',
      height: '70vh',
      width: '30vw',
    },

    triangle: {
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
      zIndex: 0,
    },

    btnRock: {
      position: 'absolute',
      bottom: '0%',
      left: '50%',
      transform: 'translate(-50%, -20%)',
      zIndex: 1,
    },

    btnPaper: {
      position: 'absolute',
      top: '0%',
      left: '0%',
      zIndex: 1,
    },

    btnScissors: {
      position: 'absolute',
      top: '0%',
      right: '0%',
      zIndex: 1,
    },
  }),
);
