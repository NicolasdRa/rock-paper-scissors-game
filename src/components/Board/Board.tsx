import React from 'react';
import { Box } from '@material-ui/core';

import { PickBtn } from '../PickBtn/PickBtn';

import triangle from '../../icons/bg-triangle.svg';

import { useStyles } from './styles';

export const Board = () => {
  const classes = useStyles();

  return (
    <Box className={classes.main}>
      <Box className={classes.btnRock}>
        <PickBtn type='rock' />
      </Box>
      <Box className={classes.btnPaper}>
        <PickBtn type='paper' />
      </Box>
      <Box className={classes.btnScissors}>
        <PickBtn type='scissors' />
      </Box>
      <Box className={classes.triangle}>
        <img src={triangle} alt='' />
      </Box>
    </Box>
  );
};
