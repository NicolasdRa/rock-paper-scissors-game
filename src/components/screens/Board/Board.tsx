import { Box } from "@material-ui/core";

import { Header } from "../../ui/Header/Header";
import { PickBtn } from "../../ui/PickBtn/PickBtn";
import triangle from "../../../icons/bg-triangle.svg";
import { Footer } from "../../ui/Footer/Footer";
import { useStyles } from "./styles";

export const Board = () => {
  const classes = useStyles();

  return (
    <Box className={classes.boardMain}>
      <Header />
      <Box className={classes.boardContent}>
        <Box className={classes.btnsContainer}>
          <Box className={classes.btnPaper}>
            <PickBtn type="paper" />
          </Box>
          <Box className={classes.btnRock}>
            <PickBtn type="rock" />
          </Box>

          <Box className={classes.btnScissors}>
            <PickBtn type="scissors" />
          </Box>
        </Box>
        <img src={triangle} alt="" className={classes.triangle} />
      </Box>
      <Footer />
    </Box>
  );
};
