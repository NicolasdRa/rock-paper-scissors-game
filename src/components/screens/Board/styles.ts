import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    boardMain: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
    },

    boardContent: {
      marginTop: "2rem",
      position: "relative",
      display: "flex",
      justifyContent: "center",

      [theme.breakpoints.down("md")]: {
        marginTop: "0",
        height: "50vh",
        width: "100vw",
      },
    },

    triangle: {
      position: "absolute",
      height: "16rem",
      top: "50%",
      left: "50%",
      transform: "translate(-50%, -50%)",
      zIndex: 0,

      [theme.breakpoints.down("md")]: {
        height: "10rem",
      },
    },

    btnsContainer: {
      display: "flex",
      height: "52vh",

      [theme.breakpoints.down("md")]: {
        marginTop: "2rem",
        height: "42vh",
      },
    },

    btnRock: {
      zIndex: 1,
      alignSelf: "flex-end",
      [theme.breakpoints.down("md")]: {
        margin: "2rem -2rem",
      },
    },

    btnPaper: {
      zIndex: 1,

      [theme.breakpoints.down("md")]: {
        marginTop: "2rem",
      },
    },

    btnScissors: {
      zIndex: 1,

      [theme.breakpoints.down("md")]: {
        marginTop: "2rem",
      },
    },
  })
);
