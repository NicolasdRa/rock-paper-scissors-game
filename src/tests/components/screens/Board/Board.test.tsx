import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Board } from "../../../../components/screens/Board/Board";

describe("tests on <Board/>", () => {
  test("should render <Board/> correctly", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Board />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
