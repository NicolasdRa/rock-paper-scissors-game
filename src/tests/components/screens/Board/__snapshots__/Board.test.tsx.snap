// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Board/> should render <Board/> correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-8 makeStyles-boardMain-1"
>
  <div
    className="MuiBox-root MuiBox-root-15 makeStyles-header-9"
  >
    <div
      className="MuiBox-root MuiBox-root-16 makeStyles-titleBox-10"
    >
      <h3
        className="MuiTypography-root makeStyles-title-11 MuiTypography-h3"
      >
        Rock
      </h3>
      <h3
        className="MuiTypography-root makeStyles-title-11 MuiTypography-h3"
      >
        Paper
      </h3>
      <h3
        className="MuiTypography-root makeStyles-title-11 MuiTypography-h3"
      >
        Scissors
      </h3>
    </div>
    <div
      className="MuiBox-root MuiBox-root-17 makeStyles-scoreBox-12"
    >
      <h3
        className="MuiTypography-root makeStyles-scoreTitle-13 MuiTypography-h3"
      >
        score
      </h3>
      <h3
        className="MuiTypography-root makeStyles-score-14 MuiTypography-h3"
      >
        0
      </h3>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-18 makeStyles-boardContent-2"
  >
    <div
      className="MuiBox-root MuiBox-root-19 makeStyles-btnsContainer-4"
    >
      <div
        className="MuiBox-root MuiBox-root-20 makeStyles-btnPaper-6"
      >
        <div
          className="MuiBox-root MuiBox-root-34 makeStyles-main-21"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root makeStyles-btn-22 makeStyles-btnPaper-29"
            href="/game"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="MuiIconButton-label"
            >
              <div
                className="MuiBox-root MuiBox-root-35 makeStyles-iconContainer-26"
              >
                <img
                  alt="pick btn"
                  className="makeStyles-icon-24"
                  src="icon-paper.svg"
                />
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-36 makeStyles-btnRock-5"
      >
        <div
          className="MuiBox-root MuiBox-root-37 makeStyles-main-21"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root makeStyles-btn-22 makeStyles-btnRock-28"
            href="/game"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="MuiIconButton-label"
            >
              <div
                className="MuiBox-root MuiBox-root-38 makeStyles-iconContainer-26"
              >
                <img
                  alt="pick btn"
                  className="makeStyles-icon-24"
                  src="icon-rock.svg"
                />
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-39 makeStyles-btnScissors-7"
      >
        <div
          className="MuiBox-root MuiBox-root-40 makeStyles-main-21"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root makeStyles-btn-22 makeStyles-btnScissors-30"
            href="/game"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="MuiIconButton-label"
            >
              <div
                className="MuiBox-root MuiBox-root-41 makeStyles-iconContainer-26"
              >
                <img
                  alt="pick btn"
                  className="makeStyles-icon-24"
                  src="icon-scissors.svg"
                />
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
    <img
      alt=""
      className="makeStyles-triangle-3"
      src="bg-triangle.svg"
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-43 makeStyles-footer-42"
  >
    <div
      className="MuiBox-root MuiBox-root-44"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-startBtn-45"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sart New Match
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiBox-root MuiBox-root-46"
    >
      <div
        className="MuiBox-root MuiBox-root-55 makeStyles-main-47"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-btn-54 MuiButton-outlinedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Rules
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
